src/app/components/core/control-dispatcher/add-control-script/add-control-script.component.ts

```diff
@@ -190,15 +191,25 @@ export class AddControlScriptComponent implements OnInit {
 
         values = Object.values(values)
           .map((v: any) => {
-            return { [v.key]: v.value };
-          }).reduce((r, c) => ({ ...r, ...c }), {})
+            return {
+              ...(v.key?.trim() && { [v.key?.trim()]: v.value?.trim() })
+            };
+          }).reduce((r, c) => ({ ...r, ...c }), {});
 
-        if ('write' in val) {
-          val['write'].values = values;
-        } else if ('operation' in val) {
-          val['operation'].parameters = values;
-        } else if ('script' in val) {
-          val['script'].parameters = values;
+        for (const key in val) {
+          if (['operation', 'script', 'write'].includes(key)) {
+            if (key !== 'write') {
+              val[key].parameters = values;
+              if (Object.keys(values).length == 0) {
+                delete val[key]?.parameters;
+              }
+              return;
```

Comment:
Are we sure there will be only one of the ['operation', 'script', 'write'] in loop?